sonic-ycabled/ycable/ycable_utilities/y_cable_helper.py

```diff
@@ -895,7 +895,11 @@ def toggle_mux_tor_direction_and_update_read_side(state, logical_port_name, phys
         helper_logger.log_error("Error: Could not get port instance for read side for while processing a toggle Y cable port {} {}".format(physical_port, threading.currentThread().getName()))
         return (-1, -1)
 
-    read_side = port_instance.get_read_side()
+    try:
+        read_side = port_instance.get_read_side()
+    except Exception as e:
+        read_side = None
+        helper_logger.log_warning("Failed to execute the get_read_side API for port {} due to {} from update_read_side".format(logical_port_name,repr(e)))
```

Comment:
So there will be a `read` before the actual `toggle` command? Is this `read` different from normal mux probe?

Reply:
No, this is a case where if read_side was not known beforehand, driver will try to get read_side again, before toggling the mux.

Reply:
normally we wont require reading read_side again,
https://github.com/sonic-net/sonic-platform-daemons/blob/534f8399afae4c601086a3f818257c55badb8a32/sonic-ycabled/ycable/ycable_utilities/y_cable_helper.py#L974
only in case it is not known we try to re read it gain